### PR TITLE
Delay load ovirtsdk4 client library

### PR DESCRIPTION
**Pull request by jrafanie**
This is a openapi gem that eager loads all files currently so it's 10+ MB to load.

Related to https://github.com/ManageIQ/manageiq-providers-ovirt/pull/667

### CODE REVIEW

File: app/models/manageiq/providers/redhat/discovery.rb [hunk 1]
```diff
@@ -18,6 +17,8 @@ def probe(ost)
           private
 
           def ovirt_exists?(host, logger = nil)
+            require 'ovirtsdk4'
```
Comment (Fryguy) [inline app/models/manageiq/providers/redhat/discovery.rb:20]: Similarly we can probably emebded the network discovery gem (in another PR of course)

**Review by agrare — APPROVED** [review]
Not for this PR but I bet we can inherit from the Ovirt parent classes for a number of these

**Comment by agrare** [comment]
> This is a openapi gem 

Is it actually an openapi gem?  I know it was a C based gem but I didn't think they had an openapi spec

**Comment by jrafanie** [comment]
> > This is a openapi gem
> 
> Is it actually an openapi gem? I know it was a C based gem but I didn't think they had an openapi spec

Thanks, I that was a typo.  The others I did recently were open api generated.  ovirtsdk4 is not.

**Pull request merged**